docs/V6-Migration-Guide.md

```diff
@@ -0,0 +1,86 @@
+```
+
+If you are upgrading from a version before 5.0.0, please see the [previous migration guides](./V5-Migration-Guide) for more details first.
+
+-   The `@dojo/cli` should be updated to version 6.0.0, along with all the commands used by the project.
+-   If your project is using `@dojo/widgets` and `@dojo/interop`, these packages also require upgrading to version 6.0.0.
+
+**Note:** The migration tool may create line lengths that violate your projects linting rules, be sure to run your linter and manually fix any linting rule violations.
+
+### Breaking Changes:
+
+#### [Update to the minimum support TypeScript version](https://github.com/dojo/framework/pull/331)
+
+As part of the enhancements made for Dojo 6, the minimum supported has changed to `3.4.5`. This is was to support the typings required by the new functional widgets and middleware.
+
+Dojo 6 should be fully compatible with `3.4.5` onwards and updating the projects dependency should satisfy Dojo, however you may find changes are required in your project code.
+
+In addition to upgrading TypeScript, the complimentary library `tslib` should be upgraded to `~1.9.1`.
+
+#### [`has`](https://github.com/dojo/framework/pull/361) and [`widget-core`](https://github.com/dojo/framework/pull/306) merged with `core`
+
+As part of Dojo 6 we have continued to re-organize `@dojo/framework`'s module structure, as a result `@dojo/framework/widget-core` and `@dojo/framework/has` have been merged with `@dojo/framework/core`.
+
+The `@dojo/cli-upgrade-app` command should automatically migrate your projects imports, please let us know if you have any issues.
+
+#### [`@dojo/framework/widget-core/d` and `@dojo/framework/widget-core/tsx` moved to `@dojo/framework/core/vdom`](https://github.com/dojo/framework/pull/360)
+
+In addition to merging `widget-core` with `core`, the `d` and `tsx` exports have been moved into `vdom`.
+
+**Note:** There is one exception, `decorate` has been moved from `d` to `util`.
+
+The `@dojo/cli-upgrade-app` command should automatically migrate your projects imports, please let us know if you have any issues.
+
+As part of Dojo 6 we have continued to re-organize `@dojo/framework`'s module structure, as a result `@dojo/framework/widget-core` and `@dojo/framework/has` have been merged with `@dojo/framework/core`.
+
+The `@dojo/cli-upgrade-app` command should automatically migrate your projects imports, please let us know if you have any issues.
+
+#### [`WNode` are no longer decorated with `bind`](https://github.com/dojo/framework/pull/290)
+
+In previous versions of Dojo virtual Dom nodes created by widgets have been decorated with additional meta data, such as `bind`. These extra properties were never meant to be exposed to the end user, however as they are added by mutating the `DNodes` it exposed these the properties inadvertently. As of Dojo 6 we no longer mutate `DNodes` to ensure that this implementation detail of the framework is exposed to the end user.
```

Comment:
```suggestion
In previous versions of Dojo, virtual DOM nodes created by widgets get decorated with additional metadata such as `bind`. These extra properties were never meant to get exposed to the end user. However, as these DOM nodes get added by mutating the `DNodes`, these properties were inadvertently exposed. As of Dojo version 6, we no longer mutate `DNodes` to ensure that this implementation detail of the framework is not exposed to the end user.
```

Reply:
I think this "However, as these DOM nodes get" should be "However, as this metadata gets"

Reply:
@dylans How about: 
```
In previous versions of Dojo, virtual DOM nodes created by widgets get decorated with additional metadata such as `bind`. These extra properties were never meant to get exposed to the end user. However, as this metadata gets added by mutating the `DNodes`, they were inadvertently exposed. As of Dojo version 6, we no longer mutate `DNodes` to ensure that this implementation detail of the framework is not exposed to the end user.
```